vlasovsolver/cpu_moments.cpp

```diff
@@ -162,152 +162,152 @@ void calculateMoments_R(
-              if (blockContainer.size() == 0) continue;
-              const Realf* data       = blockContainer.getData();
-              const Real* blockParams = blockContainer.getParameters();
-              const Real mass = getObjectWrapper().particleSpecies[popID].mass;
-              const Real charge = getObjectWrapper().particleSpecies[popID].charge;
-              const bool istestspecies = getObjectWrapper().particleSpecies[popID].isTestSpecies;
-
-              #ifdef DEBUG_MOMENTS
-              bool ok = true;
-              if (data == NULL && blockContainer.size() > 0) ok = false;
-              if (blockParams == NULL && blockContainer.size() > 0) ok = false;
-              if (ok == false) {
-                 stringstream ss;
-                 ss << "ERROR in moment calculation in " << __FILE__ << ":" << __LINE__ << endl;
-                 ss << "\t &data = " << data << "\t &blockParams = " << blockParams << endl;
-                 ss << "\t size = " << blockContainer.size() << endl;
-                 cerr << ss.str();
-                 exit(1);
-              }
-              #endif
-
-              // Temporary array where the moments for this species are accumulated
-              Real array[4];
-              for (int i=0; i<4; ++i) array[i] = 0.0;
-
-              // Calculate species' contribution to first velocity moments
-              for (vmesh::LocalID blockLID=0; blockLID<blockContainer.size(); ++blockLID) {
-                 blockVelocityFirstMoments(data+blockLID*WID3,
-                                           blockParams+blockLID*BlockParams::N_VELOCITY_BLOCK_PARAMS,
-                                           array);
-              } // for-loop over velocity blocks
-
-              // Store species' contribution to bulk velocity moments
-              Population & pop = cell->get_population(popID);
-              pop.RHO_R = array[0];
-              pop.V_R[0] = divideIfNonZero(array[1], array[0]);
-              pop.V_R[1] = divideIfNonZero(array[2], array[0]);
-              pop.V_R[2] = divideIfNonZero(array[3], array[0]);
-              
-              if (istestspecies == false ) {
-                  cell->parameters[CellParams::RHOM_R  ] += array[0]*mass;
-                  cell->parameters[CellParams::VX_R] += array[1]*mass;
-                  cell->parameters[CellParams::VY_R] += array[2]*mass;
-                  cell->parameters[CellParams::VZ_R] += array[3]*mass;
-                  cell->parameters[CellParams::RHOQ_R  ] += array[0]*charge;
-              }
-	      /* RHOQE is accumulated for test species as well. This is only used for calculating
-		 electric field due to charge imbalance in electron runs. */
-	      cell->parameters[CellParams::RHOQE_R  ] += array[0]*charge;
-
-           } // for-loop over spatial cells
-       }
+        if (getObjectWrapper().particleSpecies[popID].propagateSpecies == false) {
+	    continue;
+        }
```

Comment:
Maybe to be discussed on Flowdock instead.
A non-propagated species should still accumulate the moments into the combined moments, shouldn't it? However that could be done by using the species' pop.* moments computed once at the beginning or restart.